### ISSUE TITLE
Focus is not trapped in `Dialog` when opened

### ISSUE THREAD

**Issue by xsjcTony** [tailwindlabs/headlessui#2166]
**What package within Headless UI are you using?**

`@headlessui/react`

**What version of that package are you using?**

`v1.7.7`

**What browser are you using?**

`Chrome 108`

**Reproduction URL**

https://stackblitz.com/edit/vitejs-vite-2fput4?file=src/App.tsx

**Describe your issue**

Focus is not trapped in `Dialog` when opened, it can be tabbed to any element else even the browser itself.

The example is just exactly the same as the official one (copy and pasted). It's working in the official site but not in my local or reprod link

**Comment by bde-maze** [comment]
Up, same here. And it does the same with the Slide Overs

**Comment by bde-maze** [comment]
Adding a close button with a ref on the div around the close button worked.
(When the ref added on the button it does not work for some reasons)

```jsx
            <div ref={closeButtonRef}>
              <button type="button" onClick={() => setIsOpen(false)}>
                <XMarkIcon className="h-6 w-6 text-gray-800" />
              </button>
            </div>
            ```

**Comment by samhastings** [comment]
Just wanted to post a brief observation in case this helps someone get to the bottom of the issue: this seems to happen when there is only one focusable element inside the dialog. More than one and the focus trap seems to work fine!

**Comment by RobinMalfait** [comment]
Hey! Thank you for your bug report!
Much appreciated! 🙏

This should be fixed by #2172, and will be available in the next release.

You can already try it using:
- `npm install @headlessui/react@insiders`.
- `npm install @headlessui/vue@insiders`.

**Comment by quantumwebco** [comment]
In Vue this doesn't seem to be fixed and causes the error 
```
An invalid form control with name='name' is not focusable.
```
I have a form on the page and another form inside a dialog, when you open the dialog it tries to focus every element on the page before landing on the dialogs elements. There is more than 1 focussable element in the dialog
